sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -409,12 +409,23 @@ export class ServiceBusReceiverImpl implements ServiceBusReceiver {
       });
   }
 
-  private _createStreamingReceiver(
-    context: ConnectionContext,
-    entityPath: string,
-    options: CreateStreamingReceiverOptions
+  private async _createStreamingReceiver(
+    options: StreamingReceiverInitArgs
   ): Promise<StreamingReceiver> {
-    return StreamingReceiver.create(context, entityPath, options);
+    throwErrorIfConnectionClosed(this._context);
+    if (options.autoComplete == null) options.autoComplete = true;
+
+    this._streamingReceiver =
+      this._streamingReceiver ?? new StreamingReceiver(this._context, this.entityPath, options);
```

Comment:
When is this used? `this._streamingReceiver = this._streamingReceiver`
When the user calls `subscribe` for the second time while the first `subscribe` is still initializing the link?

Reply:
This use case is a little convoluted and TBH we should simplify it (probably in a future smaller PR). The way you end up here is:

```typescript
const subscription = receiver.subscribe()

// suspends the current streaming operation on the receiver but does 
// not close the receiver link. Useful if you want to stop receiving new messages
// but still settle existing ones.
await subscription.stop();

// restart the subscription (since we already have the link we can avoid some reinitialization)
receiver.subscribe()
```

Honestly, the last part is probably not a big enough deal to warrant carrying around all of this code. So removing it makes sense to me.

@ramya-rao-a , you see any issues with this? I'm fairly certain we talked about it before - I think this refactor just made it more obvious that it's overkill.

Reply:
@HarshaNalluru, yes this is for when we want to re-use the existing receiver link for another subscribe() call
@richardpark-msft This is fine

Reply:
This seems analogous to starting/stopping/restarting the processor.